packages/x-charts/src/ChartsSurface.tsx

```diff
@@ -35,18 +39,50 @@ export const ChartsSurface = React.forwardRef<SVGSVGElement, ChartsSurfaceProps>
       disableAxisListener = false,
       className,
       sx,
+      onClick,
       ...other
     } = props;
     const svgView = { width, height, x: 0, y: 0, ...viewBox };
 
     useAxisEvents(disableAxisListener);
 
+    const { axis, item } = React.useContext(InteractionContext);
+    const series = React.useContext(SeriesContext);
+
+    const onLineClick = (event: React.MouseEvent<SVGSVGElement, MouseEvent>) => {
+      const itemData = onClick
+        ? item && { item: series.line?.series[item.seriesId], index: axis.x?.index }
+        : null;
+
+      const seriesData = onClick
+        ? series.line?.series &&
+          Object.keys(series.line?.series).reduce((ret: object, key: string) => {
+            return {
+              ...ret,
+              [key]: series.line?.series[key].data[axis.x?.index ?? 0],
+            };
+          }, {})
+        : null;
+
+      onClick?.({ event, seriesData, itemData });
```

Comment:
?

```suggestion
      onClick?.(event, seriesData, itemData);
```

Reply:
I thought it would be nicer to destructure anything you want but I removed it.

Reply:
What's a bit weird, is that the same `onClick` applies to both the root of the svg and the items. Usually in react we would have a `onClick` on the surface, and another one on items.

So I agree with you that destructuring makes sense here
```suggestion
      onClick?.(event, { seriesData, itemData });
```

Another solution would have been to introduce `onClick` and `onSurfaceClick` to be used as follows:

```jsx
const LineChart = ({onClick , onSurfaceClick}) => {
   return <ChartsContainer onClick={onSurfaceClick}>
		{/* ... */}
		<LinePlot onClick={onClick}/>
	</ChartsContainer>
}
```

The `onSurfaceClick` would be transmitted to the container and the `onClick` would be more like the pie charts event listener. 

This would allow to [disable the axis listener](https://github.com/mui/mui-x/blob/master/packages/x-charts/src/LineChart/LineChart.tsx/#L143-L145) when not needed:

```diff
 disableAxisListener={
-  tooltip?.trigger !== 'axis' && axisHighlight?.x === 'none' && axisHighlight?.y === 'none'
+  !onSurfaceClick && tooltip?.trigger !== 'axis' && axisHighlight?.x === 'none' && axisHighlight?.y === 'none'
 }
```

Another advantage is more flexibility when using composition since you could set `onClick` only on some sub-components.

But your current solution is elegant, and has the advantage of ensuring that highlight and click are in sync.

@giladappsforce What do you think?